### PR TITLE
PYIC-4274 Extend TICF stub to return failed response code.

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Extend TICF stub to return failed response code.

### What changed
- Updated management api (/management/user/{userId}/statuscode/{statusCode})

### Issue tracking

- [PYI-4274](https://govukverify.atlassian.net/browse/PYI-4274)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: di-ipv-ticf-stub/lambdas/src/handlers/ticfHandler.ts [hunk 1]
```diff
@@ -17,9 +17,9 @@ export async function handler(
   }
 
   try {
-    const response: TicfResponse = await processGetVCRequest(ticfRequest);
+    const response: ServiceResponse = await processGetVCRequest(ticfRequest);
     console.info(`Returning ${JSON.stringify(response)}`);
-    return buildApiResponse(response);
+    return buildApiResponse(response.response, response.statusCode);
```
Comment (Joe-Edwards-GDS) [inline di-ipv-ticf-stub/lambdas/src/handlers/ticfHandler.ts]: If you wanted to make this slightly neater, you could destructure in the assignment:
```
    const { response, statusCode } = await processGetVCRequest(ticfRequest);
    console.info(`Returning ${JSON.stringify(response)}`);
    return buildApiResponse(response, statusCode);
```

File: di-ipv-ticf-stub/lambdas/src/management/handlers/managementHandler.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export async function handler(
   if (!userId) {
     return buildApiResponse({ errorMessage: "Missing userId." }, 400);
   }
-
+  const statusCode = event.pathParameters?.statusCode;
```
Comment (Joe-Edwards-GDS) [inline di-ipv-ticf-stub/lambdas/src/management/handlers/managementHandler.ts]: If you default the status code here it'll make the rest of the code neater, and I think it's the right place for it too (the fact it defaults to 200 is a property of the management API, not the storage service).

i.e. `const statusCode = event.pathParameters?.statusCode ?? 200;`

File: di-ipv-ticf-stub/lambdas/src/services/ticfService.ts [hunk 3]
```diff
@@ -49,12 +50,15 @@ export async function processGetVCRequest(
 
   const returnJwt = await signJwt(payload, ticfSigningKey);
   return {
-    sub: ticfRequest.sub,
-    govuk_signin_journey_id: ticfRequest.govuk_signin_journey_id,
-    vtr: ticfRequest.vtr,
-    vot: ticfRequest.vot,
-    vtm: ticfRequest.vtm,
-    "https://vocab.account.gov.uk/v1/credentialJWT": [returnJwt],
+    response: {
```
Comment (Joe-Edwards-GDS) [inline di-ipv-ticf-stub/lambdas/src/services/ticfService.ts:53]: I think you need to change the return type of the method to match this change.

Have you set up TS correctly in VS Code? It should warn you pretty quickly about this. I'm also a bit surprised it doesn't get flagged in the pre-merge checks, I wonder if we need to add something to `pre-merge-checks.yml` for this (perhaps as a separate ticket, if it's not quick).

**Review by Joe-Edwards-GDS — COMMENTED** [review]
Looks reasonable, a couple of comments (the type one definitely needs fixing).

File: di-ipv-ticf-stub/lambdas/src/services/ticfService.ts (hunk 3 above)
Comment (rpayal) [reply di-ipv-ticf-stub/lambdas/src/services/ticfService.ts:53]: Method (processGetVCRequest) been changed to return (Promise-ServiceResponse).

File: di-ipv-ticf-stub/lambdas/src/handlers/ticfHandler.ts (hunk 1 above)
Comment (rpayal) [reply di-ipv-ticf-stub/lambdas/src/handlers/ticfHandler.ts]: done.

File: di-ipv-ticf-stub/lambdas/src/management/handlers/managementHandler.ts (hunk 2 above)
Comment (rpayal) [reply di-ipv-ticf-stub/lambdas/src/management/handlers/managementHandler.ts]: done...bring it up to api level.

File: di-ipv-ticf-stub/lambdas/src/services/ticfService.ts (hunk 3 above)
Comment (Joe-Edwards-GDS) [reply di-ipv-ticf-stub/lambdas/src/services/ticfService.ts:53]: Sounds good (although I can't see the change yet?)

Comment (rpayal) [reply di-ipv-ticf-stub/lambdas/src/services/ticfService.ts:53]: VScode missing the autosave, now u'll see.

**Review by Joe-Edwards-GDS — APPROVED** [review]

**Pull request merged**